src/main/java/edu/wisc/my/keyvalue/controller/KeyValueStoreController.java

```diff
@@ -57,6 +66,102 @@ public void setUsernameAttr(String attr) {
         }
     }
     
+    @RequestMapping(value="/{scope}/{key}", method=RequestMethod.PUT)
+    public @ResponseBody void putScopedKeyValue(HttpServletRequest request, HttpServletResponse response, @PathVariable String scope, @PathVariable String key, @RequestBody String valueJson) throws IOException {
+      String byUser = env.getRequiredProperty("scope." + scope + ".byUser");
+      boolean scopeUserBased = Boolean.parseBoolean(byUser);
+      logger.trace("scope " + scope + " byUser? " +byUser + " : " + scopeUserBased);
```

Comment:
Consider placeholder log message format to avoid string concatenation when not running at trace log level.

`logger.trace("scope {}  byUser? {} : {}.", scope, byUser, scopeUserBased);`